tests/test-scribe.c

```diff
@@ -310,6 +311,40 @@ test_scribe_write_cb (GObject      *source,
 }
 
 static void
+assert_first_mib_zeroed (Fixture *fixture)
+{
+  gboolean ret;
+  g_autofree gchar *target_contents = NULL;
+  gsize target_length = 0;
+  g_autofree gchar *expected_contents = g_malloc0 (ONE_MIB);
+  g_autoptr(GError) error = NULL;
+
+  if (fixture->data->create_memfd)
+    {
+      g_autoptr(GInputStream) input = NULL;
+
+      target_contents = g_malloc (ONE_MIB);
+
+      seek_to_start (fixture->memfd);
+      input = g_unix_input_stream_new (fixture->memfd, /* close_fd */ FALSE);
+      ret = g_input_stream_read_all (input, target_contents, ONE_MIB,
+                                     &target_length, NULL, &error);
+    }
+  else
+    {
+      ret = g_file_get_contents (fixture->target_path,
+                                 &target_contents, &target_length,
+                                 &error);
+    }
+
+  g_assert_no_error (error);
+  g_assert (ret);
+
+  g_assert_cmpmem (expected_contents, ONE_MIB,
+                   target_contents, MIN (ONE_MIB, target_length));
```

Comment:
Nitpick: Rather than `MIN`, you might want a `g_assert_cmpuint (target_length, <=, ONE_MIB)`.

Reply:
That assertion would always fail because `target_length` should be the full length of the image. I guess `target_length, >=, ONE_MIB` would do.